### ISSUE TITLE
run_test.py exit code not checked on Windows when other commands are tested

### ISSUE THREAD

**Issue by jjhelmus** [conda/conda-build#1112]
On the win-32 (and likely win-64) platform conda-build 1.21.5 does not check the exit code of the `run_test.py` script when other commands are included in the test block.  

Minimal example:

meta.yaml
``` yaml
package:
    name: foobar
    version: 1.0

test:
    requires:
        - python
    commands:
        - python -V
```

run_test.py
``` Python
assert False
```

conda-build version 1.20.

```
C:\Users\IEUser\foobar>conda build .
Removing old build environment
BUILD START: foobar-1.0-0
Package: foobar-1.0-0
no source
number of files: 0
Fixing permissions
Fixing permissions
BUILD END: foobar-1.0-0
TEST START: foobar-1.0-0
Fetching package metadata ...........
Solving package specifications: ..........

The following NEW packages will be INSTALLED:

    foobar:         1.0-0         local
    pip:            8.1.2-py35_0
    python:         3.5.2-0
    setuptools:     23.0.0-py35_0
    vs2015_runtime: 14.0.25123-0
    wheel:          0.29.0-py35_0

Linking packages ...
[      COMPLETE      ]|##################################################| 100%

C:\Users\IEUser\Miniconda3\conda-bld\test-tmp_dir>call activate _test
===== testing package: foobar-1.0-0 =====
running run_test.py
Traceback (most recent call last):
  File "C:\Users\IEUser\Miniconda3\conda-bld\test-tmp_dir\run_test.py", line 27,
 in <module>
    assert False
AssertionError
Python 3.5.2 :: Continuum Analytics, Inc.
TEST END: foobar-1.0-0
# If you want to upload this package to anaconda.org later, type:
#
# $ anaconda upload C:\Users\IEUser\Miniconda3\conda-bld\win-32\foobar-1.0-0.tar
.bz2
#
# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes


C:\Users\IEUser\foobar>conda build --version
conda-build 1.21.5
```

This seems to be a reversion as conda build version 1.20.3 did check the exit code and the above test would fail

```
C:\Users\IEUser\foobar>conda install conda-build=1.20
Fetching package metadata .........
Solving package specifications: ..........

Package plan for installation in environment C:\Users\IEUser\Miniconda3:

The following packages will be DOWNGRADED due to dependency conflicts:

    conda-build: 1.21.5-py35_0 --> 1.20.3-py35_0

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ]|##################################################| 100%
Linking packages ...
[      COMPLETE      ]|##################################################| 100%

C:\Users\IEUser\foobar>conda build .
Removing old build environment
BUILD START: foobar-1.0-0
Package: foobar-1.0-0
no source
number of files: 0
Fixing permissions
Fixing permissions
BUILD END: foobar-1.0-0
TEST START: foobar-1.0-0
Fetching package metadata ...........

WARNING: conda-build appears to be out of date. You have version 1.20.3 but the
latest version is 1.21.5. Run

conda update -n root conda-build

to get the latest version.

Solving package specifications: ..........

The following NEW packages will be INSTALLED:

    foobar:         1.0-0         local
    pip:            8.1.2-py35_0
    python:         3.5.2-0
    setuptools:     23.0.0-py35_0
    vs2015_runtime: 14.0.25123-0
    wheel:          0.29.0-py35_0

Linking packages ...
[      COMPLETE      ]|##################################################| 100%
===== testing package: foobar-1.0-0 =====
running run_test.py
Traceback (most recent call last):
  File "C:\Users\IEUser\Miniconda3\conda-bld\test-tmp_dir\run_test.py", line 27,
 in <module>
    assert False
AssertionError
TESTS FAILED: foobar-1.0-0
```

**Comment by jjhelmus** [comment]
Should add that if you remove the `commands: -python` block the exit code of the `run_tests.py` script is checked and the tests fail.

**Comment by msarahan** [comment]
interesting.  Thanks for reporting.  I'll look into it.

**Comment by msarahan** [comment]
Closing - fixed by #1122

**Comment by jjhelmus** [comment]
Thanks for the quick fix @msarahan
